src/simple_cache_client.rs

```diff
@@ -2050,6 +2050,191 @@ impl SimpleCacheClient {
+    ///
+    /// // this will fetch the all the elements (with their scores) in the
+    /// // sorted set, sorted by score in ascending order. Any valid
+    /// // `core::ops::Range` can be used to select and limit the returned
+    /// // elements. Here we use `..` to select all the elements.
+    /// let result = momento.sorted_set_fetch_by_score(
+    ///     &cache_name,
+    ///     "test sorted set",
+    ///     Order::Ascending,
+    ///     ..,
+    /// ).await?;
+    ///
+    /// if let Ok(elements) = TryInto::<Vec<(Vec<u8>, f64)>>::try_into(result) {
+    ///     // we only set 6 elements, check that we got exactly what we set
+    ///     assert_eq!(elements.len(), test_elements.len());
+    ///
+    ///     // we can iterate and print the elements
+    ///     for (idx, (value, score)) in elements.iter().enumerate() {
+    ///         println!("value: {:?} score: {score}", value);
+    ///
+    ///         // check that the value is correct
+    ///         assert_eq!(*value, test_elements[idx].value);
+    ///     }
+    /// } else {
+    ///     panic!("sorted set was missing or the response was invalid");
+    /// }
+    ///
+    /// // by changing the range, we can get a subset of the sorted set. This
+    /// // will select just 3 elements, again in ascending order.
+    /// let result = momento.sorted_set_fetch_by_score(
+    ///     &cache_name,
+    ///     "test sorted set",
+    ///     Order::Ascending,
+    ///     3.0..=7.0,
+    /// ).await?;
+    ///
+    /// if let Ok(elements) = TryInto::<Vec<(Vec<u8>, f64)>>::try_into(result) {
+    ///     // we only wanted 3 elements, check that we got 3
+    ///     assert_eq!(elements.len(), 3);
+    ///
+    ///     // check that the values are correct
+    ///     for (idx, (value, _score)) in elements.iter().enumerate() {
+    ///         assert_eq!(*value, test_elements[idx + 1].value);
+    ///     }
+    /// } else {
+    ///     panic!("sorted set was missing or the response was invalid");
+    /// }
+    /// # Ok(())
+    /// # })
+    /// # }
+    /// ```
+    pub async fn sorted_set_fetch_by_score(
+        &mut self,
+        cache_name: &str,
+        set_name: impl IntoBytes,
+        order: sorted_set::Order,
+        range: impl RangeBounds<f64>,
+    ) -> MomentoResult<SortedSetFetch> {
+        use core::ops::Bound;
+        use sorted_set_fetch_request::by_score::{Max, Min, Score};
+        use sorted_set_fetch_request::{ByScore, Range};
+        use sorted_set_fetch_response::found::Elements;
+
+        // transforms the Rust `Range` start into a Momento min score.
+        let min = match range.start_bound() {
+            Bound::Included(v) => Min::MinScore(Score {
+                score: *v,
+                exclusive: false,
+            }),
+            Bound::Excluded(v) => Min::MinScore(Score {
+                score: *v,
+                exclusive: true,
+            }),
+            Bound::Unbounded => Min::UnboundedMin(Unbounded {}),
```

Comment:
following up on a question I asked in slack that I think I figured out the answer to; since I'm not very familiar with rust's Range type, I'm guessing that if the user wanted to express "fetch the last 5 items", and we were trying to get a start index of "undefined" and an end index of -5, they could express that via a Rust range of `..-5`?